plugins/search-react/src/context/SearchContext.tsx

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { JsonObject } from '@backstage/types';
+import { useApi, AnalyticsContext } from '@backstage/core-plugin-api';
+import { SearchResultSet } from '@backstage/plugin-search-common';
+import React, {
+  createContext,
+  PropsWithChildren,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import useAsync, { AsyncState } from 'react-use/lib/useAsync';
+import usePrevious from 'react-use/lib/usePrevious';
+import { searchApiRef } from '../api';
+
+type SearchContextValue = {
+  result: AsyncState<SearchResultSet>;
+  setTerm: React.Dispatch<React.SetStateAction<string>>;
+  setTypes: React.Dispatch<React.SetStateAction<string[]>>;
+  setFilters: React.Dispatch<React.SetStateAction<JsonObject>>;
+  setPageCursor: React.Dispatch<React.SetStateAction<string | undefined>>;
+  fetchNextPage?: React.DispatchWithoutAction;
+  fetchPreviousPage?: React.DispatchWithoutAction;
+} & SearchContextState;
+
+/**
+ *
+ * @public
+ */
+export type SearchContextState = {
+  term: string;
+  types: string[];
+  filters: JsonObject;
+  pageCursor?: string;
+};
+
+/**
+ * @public
+ */
+export const SearchContext = createContext<SearchContextValue | undefined>(
```

Comment:
Would be best if we could add a version bridge for this one right? I'm assuming this is a context that is shared pretty broadly across the app, potentially consumed by different plugins on different versions of `search-react`. Perhaps no need to rush that in today, but best if we could get it in place before 1.1? (@backstage/techdocs-core)

Reply:
Thanks @Rugvip I think I'll try to not export the context itself in the next one (when moving over more compoents) and just the provider + hooks. Do we still need a version bridge? And on that topic, I think the version bridge thing is a new thing to me, haven't heard about it before, where can I read up on it?

Reply:
Yep, the versioned context both makes it easier to evolve a context, but also makes sure that package duplication doesn't break the app. A regular React context is by-reference so it can't handle package duplication. The API itself is reasonably well documented at https://backstage.io/docs/reference/version-bridge, but ofc look at existing usage in the repo just to match error handling and so on a bit.

Reply:
Thank you, will look into it! @Rugvip

Reply:
https://github.com/backstage/backstage/pull/10808